src/borg/remote.py

```diff
@@ -358,36 +583,51 @@ def pop_preload_msgid(chunkid):
                 del self.chunkid_to_msgids[chunkid]
             return msgid
 
-        def handle_error(error, res):
-            error = error.decode('utf-8')
+        def handle_error(unpacked):
+            error = unpacked[b'exception_class'].decode()
+            old_server = b'exception_res' in unpacked
+            args = unpacked.get(b'exception_args', None)
+
             if error == 'DoesNotExist':
                 raise Repository.DoesNotExist(self.location.orig)
             elif error == 'AlreadyExists':
                 raise Repository.AlreadyExists(self.location.orig)
             elif error == 'CheckNeeded':
                 raise Repository.CheckNeeded(self.location.orig)
             elif error == 'IntegrityError':
-                raise IntegrityError(res)
+                if old_server:
+                    raise IntegrityError('<not available>')
```

Comment:
bad if that msg gets posted to some stupid pastebin that (like github reviews) interprets everything in `<...>` as html or formatting tag and makes everything unknown invisible.